src/dexpo.cpp

```diff
@@ -61,39 +61,107 @@ main ()
     {
-      usleep (1000);
-
-      if (dexpo_width == 0)
+      switch (event->response_type & ~0x80)
         {
-          // Drawing screenshots starting from the left top corner
-          auto act_width = dexpo_padding;
-          for (const auto &dexpo_pixmap : v)
+        case XCB_EXPOSE:
+          {
+            if (dexpo_width == 0)
+              {
+                // Drawing screenshots starting from the left top corner
+                auto act_width = dexpo_padding;
+                for (const auto &dexpo_pixmap : v)
+                  {
+                    xcb_copy_area (window::c_, dexpo_pixmap.id, w.id,
+                                   desktop_pixmap::gc_, 0, 0, act_width,
+                                   dexpo_padding, dexpo_pixmap.width,
+                                   dexpo_pixmap.height);
+                    act_width += dexpo_pixmap.width;
+                    act_width += dexpo_padding;
+                  };
+              }
+            else if (dexpo_height == 0)
+              {
+                auto act_height = dexpo_padding;
+                for (const auto &dexpo_pixmap : v)
+                  {
+                    xcb_copy_area (window::c_, dexpo_pixmap.id, w.id,
+                                   desktop_pixmap::gc_, 0, 0, dexpo_padding,
+                                   act_height, dexpo_pixmap.width,
+                                   dexpo_pixmap.height);
+                    act_height += dexpo_pixmap.height;
+                    act_height += dexpo_padding;
+                  };
+              }
+            w.highlight_window (0, dexpo_hlcolor);
+            break;
+          }
+        case XCB_KEY_PRESS:
+          {
+            xcb_key_press_event_t *kp
+                = reinterpret_cast<xcb_key_press_event_t *> (event);
+            ;
+            if (kp->detail == 114) // right arrow
+              {
+                highlighted += 1;
+                highlighted = highlighted % int(v.size ());
+                w.highlight_window (0,
+                                    dexpo_bgcolor); // TODO: add a function like
+                                                    // unhlight_all or smth
+                w.highlight_window (1, dexpo_bgcolor);
+                w.highlight_window (2, dexpo_bgcolor);
+                w.highlight_window (highlighted, dexpo_hlcolor);
+              }
+            if (kp->detail == 113) // left arrow
             {
-              xcb_copy_area (window::c_, dexpo_pixmap.id, w.id,
-                             desktop_pixmap::gc_, 0, 0, act_width,
-                             dexpo_padding, dexpo_pixmap.width,
-                             dexpo_pixmap.height);
-              act_width += dexpo_pixmap.width;
-              act_width += dexpo_padding;
-            };
+              if (highlighted == 0)
+                {
+                  highlighted = v.size () ;
+                }
+                  highlighted -= 1;
+            }
```

Comment:
just use mod

Reply:
are you sure -1 mod v.size() =v.size()-1?

Reply:
desktop ids start with zero. just as vector's indexes